Plugins/ConnectSwiftPlugin/ConnectClientGenerator.swift

```diff
@@ -17,26 +17,27 @@ import Foundation
 import SwiftProtobufPluginLibrary
 
 /// Responsible for generating services and RPCs that are compatible with the Connect library.
+@main
 final class ConnectClientGenerator: Generator {
-    private let visibility: String
+    private var visibility = ""
 
-    required init(_ descriptor: FileDescriptor, options: GeneratorOptions) {
-        switch options.visibility {
+    override var outputFileExtension: String {
+        ".connect.swift"
+    }
+
```

Comment:
> Is the Generator ever initialized and used on its own?

It's not, I added a comment to clarify this as well

> perhaps we could even consider putting an assertion or something in that superclass outputFileExtension implementation and specify that it must be overridden in subclasses?

This sounds good to me -- I'd rather do that than omit it in subclasses so that it's explicit in each generator. Updated